src/shared/version_op.c

```diff
@@ -442,34 +442,40 @@ os_info *get_unix_version()
         fclose(os_release);
 
         // If the OS is CentOS, try to get the version from the 'centos-release' file.
-        if (info->os_platform && strcmp(info->os_platform, "centos") == 0) {
-            regex_t regexCompiled;
-            regmatch_t match[2];
-            int match_size;
-            if (version_release = fopen("/etc/centos-release","r"), version_release){
-                os_free(info->os_version);
-                static const char *pattern = "([0-9][0-9]*\\.?[0-9]*)\\.*";
-                if (regcomp(&regexCompiled, pattern, REG_EXTENDED)) {
-                    merror_exit("Cannot compile regular expression.");
-                }
-                while (fgets(buff, sizeof(buff) - 1, version_release)) {
-                    if(regexec(&regexCompiled, buff, 2, match, 0) == 0){
-                        match_size = match[1].rm_eo - match[1].rm_so;
-                        os_malloc(match_size + 1, info->os_version);
-                        snprintf (info->os_version, match_size +1, "%.*s", match_size, buff + match[1].rm_so);
-                        break;
+        if (info->os_platform) {
+            if (strcmp(info->os_platform, "centos") == 0) {
+                regex_t regexCompiled;
+                regmatch_t match[2];
+                int match_size;
+                if (version_release = fopen("/etc/centos-release","r"), version_release){
+                    os_free(info->os_version);
+                    static const char *pattern = "([0-9][0-9]*\\.?[0-9]*)\\.*";
+                    if (regcomp(&regexCompiled, pattern, REG_EXTENDED)) {
+                        merror_exit("Cannot compile regular expression.");
                     }
+                    while (fgets(buff, sizeof(buff) - 1, version_release)) {
```

Comment:
We should be able to just use `sizeof buff` here.
>  The fgets() function shall read bytes from stream into the array
       pointed to by s until n-1 bytes are read, or a <newline> is read
       and transferred to s, or an end-of-file condition is encountered.
       A null byte shall be written immediately after the last byte read
       into the array.

Reply:
I understand the point, unfortunately this modification is not atomic, since this same use is made everywhere, it should be dealt with in another issue.

Reply:
the modification of this segment is only a nesting to avoid the double condition.